pdns/opensslsigners.cc

```diff
@@ -540,6 +541,16 @@ void OpenSSLRSADNSCryptoKeyEngine::fromISCMap(DNSKEYRecordContent& drc, std::map
 
 bool OpenSSLRSADNSCryptoKeyEngine::checkKey() const
 {
+  // When changing the bitsizes, also edit them in ::create and pdnsutil.cc
+  if ((d_algorithm == DNSSECKeeper::RSASHA1 || d_algorithm == DNSSECKeeper::RSASHA1NSEC3SHA1) && (getBits() < 512 || getBits()> 4096)) {
+    return false;
```

Comment:
to avoid duplicating these key sizes even more, add an output argument and stick an error string into that?